util/Pos/Util/Util.hs

```diff
@@ -267,3 +274,7 @@ median l = NE.sort l NE.!! middle
 -}
 sleep :: MonadIO m => NominalDiffTime -> m ()
 sleep n = liftIO (threadDelay (truncate (n * 10^(6::Int))))
+
+-- | Formats two values as first and last elements of a list
+buildListBounds :: (Buildable a, Buildable b) => F.Format r (a -> b -> r)
+buildListBounds = "[" F.% F.build F.% ".." F.% F.build F.% "]"
```

Comment:
I'm not really sure about `b /= a` cases.

Reply:
Do you mean `a == b` case?

Reply:
Ah, you mean types, not values. I mean it prints a strange thing when value of `a` equals to the value of `b`.

Reply:
>print single thing only.

I'm fairly sure you can't change amount of things you want to print with `Format` alone. You'd have to have an actual printing function that has such logic inside it.